admin-dev/themes/new-theme/js/components/grid/extension/ajax-bulk-action-extension.ts

```diff
@@ -0,0 +1,193 @@
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+import {Grid} from '@PSTypes/grid';
+import ProgressModal from '@components/modal/progress-modal';
+import ConfirmModal from '@components/modal/confirm-modal';
+import GridMap from '@components/grid/grid-map';
+import Router from '@components/router';
+
+import ClickEvent = JQuery.ClickEvent;
+
+const {$} = window;
+
+/**
+ * Handles submit of grid actions
+ */
+export default class AjaxBulkActionExtension {
+  private router = new Router();
+
+  /**
+   * Extend grid with bulk action submitting
+   *
+   * @param {Grid} grid
+   */
+  extend(grid: Grid): void {
+    grid
+      .getContainer()
+      .on('click', GridMap.bulks.ajaxAction, (event: ClickEvent) => {
+        const $ajaxButton: JQuery<HTMLInputElement> = $<HTMLInputElement>(event.currentTarget);
+        const $checkboxes: JQuery<HTMLInputElement> = $<HTMLInputElement>(GridMap.bulks.checkedCheckbox);
+        const selectedIds: string[] = $checkboxes.get().map((checkbox: HTMLInputElement) => checkbox.value);
+
+        if (selectedIds.length === 0) {
+          return;
+        }
+
+        const confirmBulkAction = $ajaxButton.data('confirmBulkAction') ?? true;
+
+        if (confirmBulkAction) {
+          const progressionTitle = $ajaxButton.data('progressTitle');
+          const closeButtonLabel = $ajaxButton.data('cancelLabel') || 'Cancel';
+          const confirmTitle = $ajaxButton.data('confirmTitle') || 'Apply modifications';
```

Comment:
Need translations

Reply:
Not in these ones, these are the default fallback to avoid having empty strings The translated and contextualized wording are injected via the data attributes